docs/build/apps/example-application-tutorial/manage-trust.mdx

```diff
@@ -66,14 +66,68 @@ The layout of the page is quite similar to our contacts page. It has a table dis
   import { invalidateAll } from "$app/navigation";
 
   // We import some of our `$lib` functions
-  import { submit } from "$lib/stellar/horizonQueries";
   import { createChangeTrustTransaction } from "$lib/stellar/transactions";
   import { fetchAssets } from "$lib/utils/stellarExpert";
 
+  // We import the Server and API types from @stellar/stellar-sdk/rpc for transaction submission
+  import { Server, Api, LinearSleepStrategy } from "@stellar/stellar-sdk/rpc";
+  import { xdr } from "@stellar/stellar-sdk";
+  import { decode } from "@stellar/stellar-xdr-json";
+  import { error } from "@sveltejs/kit";
+
   // The `open` Svelte context is used to open the confirmation modal
   import { getContext } from "svelte";
   const { open } = getContext("simple-modal");
 
+  // RPC server configuration for transaction submission
+  const rpcUrl = "https://soroban-testnet.stellar.org";
+  const server = new Server(rpcUrl);
+
+  // Local submit function to send transactions using RPC
+  const submit = async (transaction) => {
+    try {
+      // Send the transaction and get initial response
+      const sendTransactionResponse = await server.sendTransaction(transaction);
+
+      // Check if transaction was accepted for processing
+      if (sendTransactionResponse.status !== "PENDING") {
+        throw new Error(
+          `Failed to send transaction, status: ${sendTransactionResponse.status}`,
+        );
+      }
+
+      // Poll for the final transaction status using LinearSleepStrategy
+      const finalResponse = await server.pollTransaction(
+        sendTransactionResponse.hash,
+      );
+
+      // Handle the final transaction status
+      switch (finalResponse.status) {
+        case Api.GetTransactionStatus.FAILED:
+          // Include decoded resultXdr for failed transactions
+          let errorDetails = `Transaction failed with status: ${finalResponse.status}`;
+          if (finalResponse.resultXdr) {
+            errorDetails += `\nResult XDR: ${finalResponse.resultXdr}`;
```

Comment:
the SDK decodes into XDR structures for you, so it'd just be `[object Object]` if you printed it out here. Need to re-encode for it to be useful as a string :+1: 

```suggestion
            errorDetails += `\nResult XDR: ${finalResponse.resultXdr.toXDR("base64")}`;
```